bpf/lib/common.h

```diff
@@ -342,14 +342,22 @@ struct endpoint_info {
 	__u32		pad[2];
 };
 
+#define DIRECTION_EGRESS 0
+#define DIRECTION_INGRESS 1
+
 struct edt_id {
 	__u64		id;
+	__u8		direction;
+	__u8		pad[7];
```

Comment:
Given that the `id` field is always(?) set to a 32-bit value, it would be better redefine this as:
```
struct edt_id {
	__u32		id;
	__u8		direction;
	__u8		pad[3];
};
```
This would also keep the size of `edt_id` intact. This is nice since many map implementations may have optimized hashing of 32 or 64-bit keys (not sure if applies to bpf maps, though).

Corresponding change would need to be done on Go `type EdtId` as well.

Looks like in userspace the `id` is only 16 bits, for all this would fit into a 32-bit key in kernel as well.

Reply:
Good call, I've made that change.